src/components/logs-viewer/logs-viewer.tsx

```diff
@@ -43,6 +43,18 @@ export class LogsViewer extends React.Component<LogsViewerProps> {
         this.terminal.loadAddon(this.fitAddon);
         this.terminal.open(container);
         this.fitAddon.fit();
+        this.terminal.attachCustomKeyEventHandler((arg) => {
+            if (arg.ctrlKey && arg.code === "KeyC" && arg.type === "keydown") {
+                const selection = this.terminal.getSelection();
+                if (selection) {
+                    if (navigator.clipboard) {
+                        navigator.clipboard.writeText(selection);
+                    }
+                    return false;
+                }
+            }
+            return true;
+        });
```

Comment:
let's invert this conditional for less nesting and less code complexity, similar to https://github.com/argoproj/argo-workflows/pull/11596, https://github.com/argoproj/argo-workflows/pull/12146, https://github.com/argoproj/argo-workflows/pull/13094, etc.

We can also simplify some pieces as this function returns `void`, such as with optional chaining.
Also `e` or `ev` is the common convention for event names.
And this could use a comment

```suggestion
        // handle Ctrl+C for copying logs
        this.terminal.attachCustomKeyEventHandler((ev) => {
            if (ev.ctrlKey && ev.code === "KeyC" && ev.type === "keydown") {
                const selection = this.terminal.getSelection();
                if (!selection) return;
                
                return navigator.clipboard?.writeText(selection);
            }
        });
```

Reply:
Oh woops, the [`xterm` docs](https://xtermjs.org/docs/api/terminal/classes/terminal/#attachcustomkeyeventhandler) are a bit confusing as they actually didn't give _what_ the function should return

I double-checked against [the typings](https://github.com/xtermjs/xterm.js/blob/9ba6c00a195c95fcf8292a2b9084d91450e5daae/typings/xterm.d.ts#L1019) and apparently it does take a boolean, for whether the event should continue propagating or not to `xterm`. We don't really care about that part though as we're only using it as a log UI.

Reply:
Thanks for your review @agilgur5. Yes I did some nesting to handle the boolean returned value in case of selected text. But if we don't care about that as we are only logging then I'll implement your suggestion.

Reply:
The boolean return is irrelevant for our use-case, but your version is more correct so sounds good.

Also I am pretty sure we can replace this component with something much smaller (per https://github.com/argoproj/argo-workflows/issues/12059, it's quite big) because we don't use even half of its features